dotCMS/src/main/java/com/dotcms/publisher/util/PushCategoryUtil.java

```diff
@@ -5,20 +5,56 @@
 import java.io.FileNotFoundException;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+
 import com.dotcms.publisher.pusher.wrapper.CategoryWrapper;
+import com.dotcms.publishing.DotPublishingException;
 import com.dotcms.repackage.com.thoughtworks.xstream.XStream;
 import com.dotcms.repackage.com.thoughtworks.xstream.io.xml.DomDriver;
 import static com.dotcms.enterprise.publishing.remote.bundler.CategoryBundler.CATEGORY_EXTENSION;
 
 public class PushCategoryUtil {
-	
-	private Collection<File> categories;
+
 	private XStream xstream;
+
+	private Map<String, File> categoriesByInode;
+	private Set<String> categoriesTopLevel;
+
+	private int categoriesCount;
 	
-	public PushCategoryUtil(Collection<File> categories){
-		this.categories = categories;
+	public PushCategoryUtil(Collection<File> categories) throws DotPublishingException {
 		xstream=new XStream(new DomDriver());
+
+		categoriesByInode = new HashMap<String, File>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)